code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,648 @@
+                    type: array
+                    items:
+                      $ref:  '#/components/schemas/ProvisionedApp'
+                minItems: 1                 
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '409':
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorInfo'
+              example:
+                status: 409
+                code: CONFLICT
+                message: "Application already provisioned"
+        '500':
+          $ref: '#/components/responses/500'
+        '501':
+          $ref: '#/components/responses/501'
+        '503':
+          $ref: '#/components/responses/503'
+ 
+  /app/{appId}:
+    delete:
+      tags:
+        - Applications
+      summary: Terminate an Application
+      description: Terminate the running instance of an application in one or several Edge Application Servers
+      operationId: deleteApplication
+      parameters:
+        - name: appId
+          in: query
+          description: Identificator of the application generated by the OP NBI once the provisioning was successful, terminate all the instances related to this appId
+          required: false
+          schema:
+            $ref: "#/components/schemas/AppId"
+        - name: appInstanceId
+          in: query
+          description: Identificator of the specific application instance that will be terminated
+          required: false
+          schema:
+            $ref: "#/components/schemas/AppInstanceId"
+      responses:
+        '204':
+          description: Application deleted      
+        '400':
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '500':
+          $ref: '#/components/responses/500'
+        '503':
+          $ref: '#/components/responses/503'
+ 
+  /artifact:
+    post:
+      tags:  
+        - Application Artifacts 
+      summary: Uploads artifact  on an OP.                       
+      description: Artifact is a zip file containing  scripts and/or packaging files like Terraform or Helm which are required to create an instance of an application.
+      operationId: uploadArtifact
+      requestBody:
+        description: An application can consist of multiple components. App providers are allowed to define seperate artifacts for each component  
+                            or  they could define a consolidated artifact at application level. 
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ArtifactDetails'
+        required: true
+      responses:
+        '201':
```

Comment:
we may have HTTP 202 also added as this operation may not be an immediate operation. And a GET method defined to retrieve the actual result later.

Reply:
Artifact no longer exists in v0.2.0 included the 202 response for the rest of the DELETE methods

Reply:
Still I think we should have 202 supported. As app submission still may not be immediate handling.